include/windows_customizations.h

```diff
@@ -4,9 +4,13 @@
 #pragma once
 
 #ifdef _WINDOWS
+
+#ifdef _WINDLL
 #define DISKANN_DLLEXPORT __declspec(dllexport)
-#define DISKANN_DLLIMPORT __declspec(dllimport)
+#else
+#define DISKANN_DLLEXPORT __declspec(dllimport)
```

Comment:
Why are we defining DISKANN_DLLEXPORT to be (dllimport)?

Reply:
When compiling the DLL itself, it's declared as `dllexport` to make it part of the DLL interface.

When it's being used by DLL clients, it's declared as `dllimport` so that the client can access the data reference in the DLL (for data access) and that a compiler can emit a slightly more efficient function calling code (for method invocation).

Ideally, this would be called either `DISKANN_DLL_DECL` or `DISKANN_DLLEXPORT_IMPORT` or similar. Then it wouldn't be confusing why is `DISKANN_DLLEXPORT` defined as `dllimport`, but I didn't want to make too many changes with this PR, it can be done separately if needed.

(The MSDN documentation on this is [here](https://docs.microsoft.com/en-us/cpp/cpp/dllexport-dllimport) BTW)